### PR TITLE
vstack vs hstack

### LINKED ISSUE #87: default vstack in _conclude() fails with scalar per frame data
### Expected behaviour
I can just follow the typical example to make analysis from a single frame function such as 
```python
    import MDAnalysis as mda
    import MDAnalysisData
    data = MDAnalysisData.datasets.fetch_nhaa_equilibrium()

    u = mda.Universe(data.topology, data.trajectory)
    protein = u.select_atoms('protein')

    def rgyr(ag):
        return  ag.radius_of_gyration()


    import pmda.custom
    parallel_rgyr = pmda.custom.AnalysisFromFunction(rgyr, u, protein)

    parallel_rgyr.run(n_jobs=4, n_blocks=4)
    print(parallel_rgyr.results)
```

### Actual behaviour
The above fails with 

```
--------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-30-27c1019c5274> in <module>
----> 1 parallel_rgyr.run(n_blocks=4)

~/MDAnalysis/pmda/pmda/parallel.py in run(self, start, stop, step, n_jobs, n_blocks)
    374             with timeit() as conclude:
    375                 self._results = np.asarray([el[0] for el in res])
--> 376                 self._conclude()
    377 
    378         self.timing = Timing(

~/MDAnalysis/pmda/pmda/custom.py in _conclude(self)
    101 
    102     def _conclude(self):
--> 103         self.results = np.vstack(self._results)
    104 
    105 

~/anaconda3/envs/pmda/lib/python3.6/site-packages/numpy/core/shape_base.py in vstack(tup)
    281     """
    282     _warn_for_nonsequence(tup)
--> 283     return _nx.concatenate([atleast_2d(_m) for _m in tup], 0)
    284 
    285 

ValueError: all the input array dimensions except for the concatenation axis must match exactly
```

### Code to reproduce the behaviour

See above.

The problem is that the arrays in `_results` do not have the same lengths
```python
>>> [len(a) for a in parallel_rgyr._results]
[1251, 1250, 1250, 1250]
```
and `np.vstack()` does not like this. Instead one would need `np.hstack()`. However, as soon as the return value of the function is a tuple, `vstack()` works as expected.

### Currently version of MDAnalysis:
- PMDA 0.2.1
- MDAnalysis 0.19.2
- dask 1.2.2

### PR DESCRIPTION
**Pull request by orbeckst**
Fixes #87 

Changes made in this Pull Request:
 - tests
 - use concatenate instead of vstack

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by orbeckst** [comment]
Only tests so far, I don't have a clear idea how to solve this problem (see #87) and the tests.

**Comment by orbeckst** [comment]
The branch name is misleading: just using `np.hstack()` is not the solution.

**Comment by orbeckst** [comment]
@VOD555 can you add to this PR please?

Possibly also add a test for more complicated return values of the tuple (something like `(scalar, array)` and `(array, array)` and `(array,)`)

File: pmda/test/test_custom.py [hunk 1]
```diff
@@ -18,26 +18,38 @@
 from pmda import custom
 
 
-def custom_function(mobile):
+@pytest.fixture
+def universe():
+    return mda.Universe(PSF, DCD)
+
```
Comment (orbeckst) [inline pmda/test/test_custom.py]: add a `custom_function_tensor()` that uses `ag.moment_of_inertia()`.

**Comment by VOD555** [comment]
All tests passed. Merge the pull request.

**Pull request merged**